lib/jellyfish/component/file.ex

```diff
@@ -0,0 +1,89 @@
+defmodule Jellyfish.Component.File do
+  @moduledoc """
+  Module representing the File component.
+  """
+
+  @behaviour Jellyfish.Endpoint.Config
+
+  alias ExSDP.Attribute.FMTP
+  alias JellyfishWeb.ApiSpec.Component.File.Options
+  alias Membrane.RTC.Engine.Endpoint.File, as: FileEndpoint
+
+  @type properties :: %{}
+
+  @files_location "file_component_sources"
+
+  @impl true
+  def config(%{engine_pid: engine} = options) do
+    options = Map.drop(options, [:engine_pid, :room_id])
+
+    with {:ok, valid_opts} <- OpenApiSpex.cast_value(options, Options.schema()),
+         :ok <- validate_file_path(valid_opts.filePath),
+         path = expand_file_path(valid_opts.filePath),
+         {:ok, track_config} <- get_track_config(path) do
+      endpoint_spec =
+        %FileEndpoint{
+          rtc_engine: engine,
+          file_path: path,
+          track_config: track_config,
+          payload_type: track_config.fmtp.pt
+        }
+
+      {:ok, %{endpoint: endpoint_spec, properties: %{}}}
+    else
+      {:error, _reason} = error -> error
+    end
+  end
+
+  defp validate_file_path(file_path) do
+    base_path =
+      Application.fetch_env!(:jellyfish, :media_files_path)
+      |> Path.join(@files_location)
+      |> Path.expand()
+
+    file_path = expand_file_path(file_path)
+
+    cond do
+      not path_inside_directory?(file_path, base_path) -> {:error, :invalid_file_path}
+      not File.exists?(file_path) -> {:error, :file_does_not_exist}
+      true -> :ok
+    end
+  end
+
+  defp path_inside_directory?(path, directory) do
+    relative_to = Path.relative_to(path, directory)
+    String.first(relative_to) != "/"
+  end
```

Comment:
There is one edge case that is not handled properly: "." in this case, this function will return true even though it's not "inside" the directory.
I implemented a similar function in this PR: https://github.com/jellyfish-dev/jellyfish/pull/127